core/shared/src/main/scala/fs2/Scheduler.scala

```diff
@@ -114,6 +114,54 @@ abstract class Scheduler {
     sleep(d).drain
 
   /**
+   * Retries `fa` on failure, returning a single element
+   * stream with the result of `fa` as soon as it succeeds.
+   *
+   * @param delay Duration of delay before the first retry
+   *
+   * @param nextDelay Applied to the previous delay to compute the
+   *                  next, e.g. to implement exponential backoff
+   *
+   * @param maxRetries Number of attempts before failing with the
+   *                   latest error, if `fa` never succeeds
+   *
+   * @param retriable Function to determine whether a failure is
+   *                  retriable or not, defaults to retry every
+   *                  `NonFatal`. A failed stream is immediately
+   *                  returned when a non-retriable failure is
+   *                  encountered
+   */
+  def retry[F[_], A](fa: F[A],
```

Comment:
Based on where the conversation in PR left off, I think we should change `fa` to `Stream[F,A]` here and then maybe add an overload for `F[A]`.